e2e/cypress/integration/messaging/receive_message_on_socket_reconnect_spec.js

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+import users from '../../fixtures/users.json';
+
+let townsquareChannelId;
+const websockets = [];
+
+describe('Messaging', () => {
+    before(() => {
+        // # Wrap websocket to be able to connect and close connections on demand
+        cy.mockWebsockets(websockets);
+
+        // # Login and go to /
+        cy.apiLogin('sysadmin');
+        cy.visit('/');
+
+        // # Get ChannelID to use later
+        cy.getCurrentChannelId().then((id) => {
+            townsquareChannelId = id;
+        });
+    });
+
+    it('M18682-RHS fetches messages on socket reconnect when a different channel is in center', () => {
+        // # Connect all sockets
+        websockets.forEach((value) => {
+            value.connect();
+        });
+
+        // # Post a message as another user
+        cy.postMessageAs({sender: users['user-1'], message: 'abc', channelId: townsquareChannelId}).wait(TIMEOUTS.SMALL);
+
+        // # Click "Reply"
+        cy.getLastPostId().then((rootPostId) => {
+            cy.clickPostCommentIcon(rootPostId);
+
+            // # Post a message
+            cy.postMessageReplyInRHS('def');
+
+            // # Save our post for later use
+            cy.getLastPostIdRHS().as('myPostId');
+
+            // # Change channel
+            cy.get('#sidebarItem_suscipit-4').click({force: true}).then(() => {
```

Comment:
This failed on fresh install.  I think the safe bet is to off-topic channel.
```suggestion
            cy.get('#sidebarItem_off-topic').click({force: true}).then(() => {
```

Reply:
Is it possible we have different test data? For me, sysadmin on test-data starts joined to the sint channel, but it is not present on offtopic channel. Therefore, making this change make it stop working on my side.

Reply:
Yeah, looks like only me.  I'm good as is.